drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -225,13 +360,11 @@ local function do_configure(driver, device)
   local ac_filter_eps = embedded_cluster_utils.get_endpoints(device, clusters.ActivatedCarbonFilterMonitoring.ID)
   local device_type = get_device_type(driver, device)
   local profile_name = "thermostat"
-  --Note: we have not encountered thermostats with multiple endpoints that support the Thermostat cluster
   if device_type == RAC_DEVICE_TYPE_ID then
     device.log.warn_with({hub_logs=true}, "Room Air Conditioner supports only one profile")
   elseif device_type == FAN_DEVICE_TYPE_ID then
     device.log.warn_with({hub_logs=true}, "Fan supports only one profile")
-  elseif device_type == AP_DEVICE_TYPE_ID then
-    -- currently no profile switching for Air Purifier
+  elseif device_type == AP_DEVICE_TYPE_ID and #thermo_eps < 1 then
```

Comment:
I think we need to combine both the `AP_DEVICE_TYPE_ID` conditionals so that air purifier devices without thermostats can still get the air measurement support. For example, [this device](https://www.aidot.com/aidot-welov-matter-p200-pro-smart-air-purifier-for-homeup-to-1570-ft-with-air-quality-monitor.html?srsltid=AfmBOoohHvHFESMgWX8FIGGJj3Ij5mfKcH91ipvT9upptW9llLnSUJIKLw8) does not have a thermostat but still has some measurement capabilities.

I think combining them should be pretty simple, you can just add an extra check in within the combined `elseif device_type == AP_DEVICE_TYPE_ID` conditional to check if `#thermo_eps > 0` and then add the appropriate thermo components to the profile name. You can keep the thermo code you already have in the conditional below but just add this extra check before going through the thermo components.

Reply:
should be working now in line with your suggestion here :+1: